Public/Src/App/Bxl/ServerDeployment.cs

```diff
@@ -221,28 +221,33 @@ private static Possible<Unit> TryKillProcess(string assemblyFullPath)
                     || e is SystemException)
                 {
                     return new Failure<string>(I($"Failed to kill process with name '{processName}' (process id: {processToKill.Id}) and path '{assemblyFullPath}'"), new Failure<Exception>(e));
-                }
+                } 
             }
 
             return Unit.Void;
         }
 
-        /// <summary>
-        /// Calculates a hash of the contents of the BuildXL binaries from the server deployment directory.
-        /// </summary>
-        public static string GetDeploymentCacheHash(string deploymentDir)
-        {
-            AppDeployment serverDeployment = AppDeployment.ReadDeploymentManifest(deploymentDir, AppDeployment.ServerDeploymentManifestFileName);
-            return serverDeployment.TimestampBasedHash.ToHex();
-        }
-
         /// <summary>
         /// Checks if the deployed server bits are still up-to-date.
         /// </summary>
         public static bool IsServerDeploymentOutOfSync(string serverDeploymentRoot, AppDeployment clientApp, out string deploymentDir)
         {
             deploymentDir = ComputeDeploymentDir(serverDeploymentRoot);
-            return !Directory.Exists(deploymentDir) || clientApp.TimestampBasedHash.ToHex() != GetDeploymentCacheHash(deploymentDir);
+            if (Directory.Exists(deploymentDir))
+            {
+                try
+                {
+                    // Calculates a hash of the contents of the BuildXL binaries from the server deployment directory.
+                    AppDeployment serverDeployment = AppDeployment.ReadDeploymentManifest(deploymentDir, AppDeployment.ServerDeploymentManifestFileName);
+                    return clientApp.TimestampBasedHash.ToHex() != serverDeployment.TimestampBasedHash.ToHex();
+                }
+                catch (BuildXLException)

```

Comment:
Any comment on why this exception might happen?  And if it does happen, we should probably at least log a warning.